pubsub.go

```diff
@@ -799,7 +802,9 @@ func (p *PubSub) processLoop(ctx context.Context) {
 				continue
 			}
 
-			p.rt.AddPeer(pid, s.Protocol())
+			helloPacket := p.getHelloPacket()
+			p.rt.AddPeer(pid, s.Protocol(), helloPacket)
```

Comment:
After the change I mentioned in https://github.com/libp2p/go-libp2p-pubsub/pull/630/files#r2306618920, this can be
```suggestion
			if extensions := p.rt.AddPeer(pid, s.Protocol()); extensions != nil {
						if helloPacket.Control == nil {
									helloPacket.Control = &pb.ControlMessage{}
						}
						hellPacket.Control.Extensions = extensions
			}
```